python/selfie-lib/selfie_lib/Literals.py

```diff
@@ -71,6 +72,174 @@ def parse(self, string: str, language: Language) -> int:
         return int(string.replace("_", ""))
 
 
+TRIPLE_QUOTE = '"""'
+
+
+class LiteralString(LiteralFormat[str]):
+    def encode(
+        self, value: str, language: Language, encoding_policy: EscapeLeadingWhitespace
+    ) -> str:
+        if "/n" not in value:
+            if language == Language.PYTHON:
+                return self._encodeSinglePython(value)
+        else:
+            if language == Language.PYTHON:
+                return self.encodeMultiPython(value, encoding_policy)
+
+    def parse(self, string: str, language: Language) -> str:
+        if not string.startswith(TRIPLE_QUOTE):
+            if language == Language.PYTHON:
+                return self._parseSinglePython(string)
+        else:
+            if language == Language.PYTHON:
+                return self.parseMultiPython(string)
+
+    def _encodeSinglePython(self, value: str) -> str:
+        source = io.StringIO()
+        source.write('"')
+        for char in value:
+            if char == "\b":
+                source.write("\\b")
+            elif char == "\n":
+                source.write("\\n")
+            elif char == "\r":
+                source.write("\\r")
+            elif char == "\t":
+                source.write("\\t")
+            elif char == '"':
+                source.write('\\"')
+            elif char == "\\":
+                source.write("\\\\")
+            elif self._is_control_char(char):
+                source.write("\\u" + str(ord(char)).zfill(4))
+            else:
+                source.write(char)
+        source.write('"')
+        return source.getvalue()
+
+    def _is_control_char(self, c: str) -> bool:
+        return c in "\u0000\u001f" or c == "\u007f"
+
+    # combined logic from parseSingleJava and parseSingleJavaish
+    def _parseSinglePython(self, source_with_quotes: str) -> str:
+        assert source_with_quotes.startswith('"')
+        assert source_with_quotes.endswith('"')
+        source = source_with_quotes[1:-1]
+        to_unescape = self.inline_backslashes(source)  # changed from inline_dollar
+        return self._unescape_python(to_unescape)
+
+    def encodeMultiPython(
+        self, arg: str, escape_leading_whitespace: EscapeLeadingWhitespace
+    ) -> str:
+        escape_backslashes = arg.replace("\\", "\\\\")
+        escape_triple_quotes = escape_backslashes.replace(TRIPLE_QUOTE, '\\"\\"\\"')
+
+        def protect_trailing_whitespace(line):
+            if line.endswith(" "):
+                return line[:-1] + "\x20"
```

Comment:
On line 207 of this file, we handle decoding `\u` literals. But we don't handle decoding `\x` literals. We have two choices:

- change the `\x20` to `\u0020` (means the same thing, tiny bit longer)
- add `\x` decoding near line 207

Your choice, sticking with `\u` only seems easier.